frontend/packages/topology/src/components/export-app/ExportApplication.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import { ToolbarItem, Button, AlertVariant } from '@patternfly/react-core';
+import { useTranslation, Trans } from 'react-i18next';
+import { useAccessReview } from '@console/internal/components/utils';
+import { dateTimeFormatter } from '@console/internal/components/utils/datetime';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import {
+  k8sCreate,
+  k8sKill,
+  K8sResourceKind,
+  WatchK8sResource,
+} from '@console/internal/module/k8s';
+import {
+  useFlag,
+  useIsMobile,
+  USERSETTINGS_PREFIX,
+  useToast,
+  useUserSettings,
+} from '@console/shared/src';
+import { ALLOW_EXPORT_APP, EXPORT_CR_NAME } from '../../const';
+import { ExportModel } from '../../models';
+import { getExportAppData } from '../../utils/export-app-utils';
+import exportApplicationModal from './export-app-modal';
+import { exportAppUserSettings } from './types';
+
+type ExportApplicationProps = {
+  namespace: string;
+  isDisabled: boolean;
+};
+
+const ExportApplication: React.FC<ExportApplicationProps> = ({ namespace, isDisabled }) => {
+  const { t } = useTranslation();
+  const [isCreating, setIsCreating] = React.useState<boolean>(false);
+  const isMobile = useIsMobile();
+  const isExportAppAllowed = useFlag(ALLOW_EXPORT_APP);
+  const canExportApp = useAccessReview({
+    group: ExportModel.apiGroup,
+    resource: ExportModel.plural,
+    verb: 'create',
+    namespace,
+  });
+  const toast = useToast();
+  const [exportAppToast, setExportAppToast] = useUserSettings<exportAppUserSettings>(
+    `${USERSETTINGS_PREFIX}.exportApp`,
+    {},
+    true,
+  );
+
+  const exportAppResource: WatchK8sResource = React.useMemo(
+    () => ({
+      kind: ExportModel.kind,
+      name: EXPORT_CR_NAME,
+      namespace,
+      namespaced: true,
+      isList: false,
+      optional: true,
+    }),
+    [namespace],
+  );
+
+  const [exportRes, exportLoaded, exportLoadErr] = useK8sWatchResource<K8sResourceKind>(
+    exportAppResource,
+  );
```

Comment:
Can be more lazy about checking the status of the export resource such that we only do the check when the user clicks the button and not constantly watch the resource while the button is on screen?

Reply:
updated